ludwig/features/image_feature.py

```diff
@@ -287,9 +287,19 @@ def _finalize_preprocessing_parameters(
             user_specified_num_channels = False
             if preprocessing_parameters[INFER_IMAGE_DIMENSIONS]:
                 user_specified_num_channels = True
-                # Use the maximum num_channels found across all sampled images. torchvision has built-in support for
-                # upsampling images.
-                num_channels = max(num_channels_in_image(x) for x in inferred_sample)
+                channels_in_sample = np.array([num_channels_in_image(x) for x in inferred_sample])
+                if sum(channels_in_sample == 1) > len(inferred_sample) / 2:
+                    # If the majority of images in sample are 1 channel, use 1.
+                    num_channels = 1
+                elif sum(channels_in_sample == 2) > len(inferred_sample) / 2:
+                    # If the majority of images in sample are 2 channel, use 2.
+                    num_channels = 2
```

Comment:
What is the expected behavior of the image preprocessing step if the majority of images in the sample have 2 channels, but the model comes across an image with 3 channels?

My understanding was that we only cropped channels if num_channels == 1. In that case, we convert it to grayscale: https://github.com/ludwig-ai/ludwig/blob/aacabd3f5406701ecc336692c27c487ed043517c/ludwig/features/image_feature.py#L158

Not sure how we would deal with it otherwise. This is perhaps why we currently take the `max` number of channels (because padding is relatively safe operation, whereas dropping channels is not).

Let me know what you think– I might be missing some context here.

Reply:
@geoffreyangus If I understand this right, you are describing a situation where the majority of images are 2 channels (grayscale + alpha), and we come across an image with 3 channels (RGB)?

In this situation, dropping the RGB image (along with a warning) is the behavior I would expect.  What do you think the expected behavior should be?

Reply:
Good ideas.  I'll integrate these changes.  I also think you bring up a generally good idea:  giving users config advice based on the data.  Maybe we should even designate a special log category for this so we can surface this type of useful feedback in predibase.  I'm going to think about this and maybe open a new issue for it, meanwhile I'll use `logging.info`.

Reply:
Implemented Counter-based checks and logging.  

On my twitter bots example, the output now reads:

```
Inferring num_channels from the first 100 images for column profile_image_path.
  images with 1 channels: 4
  images with 3 channels: 88
  images with 4 channels: 8
Using 3 channels because it is the majority in sample. If an image with a different depth is read, will attempt to convert to 3 channels.
To explicitly set the number of channels, define num_channels in the preprocessing dictionary of the image input feature config.
```